File: src/routes/textGeneration.ts
import { Router, type Request, type Response } from "express";
import { pipelineFactory } from "../utils/pipelineFactory";
import type { TextGenerationRequest, PipelineResult } from "../types/index";

const router = Router();

router.post("/", async (req: Request, res: Response) => {
  try {
    const { text, generation }: TextGenerationRequest = req.body;

    if (!text) {
      return res.status(400).json({ error: "Missing required field: text" });
    }

    const pipe = await pipelineFactory.getPipeline("text-generation");
    const result = await pipe(text, generation);

    const response: PipelineResult = {
      success: true,
      data: result,
    };

    res.json(response);
  } catch (error: any) {
    res.status(500).json({
      success: false,
      error: error.message || "Text generation failed",
    });
  }
});

export default router;

File: python-src/routes/text_generation.py
from fastapi import APIRouter
from models.types import TextGenerationRequest, PipelineResult
from utils.pipeline_factory import pipeline_factory

router = APIRouter()

@router.post("/", response_model=PipelineResult)
async def generate_text(request: TextGenerationRequest):
    # TODO: Implement text generation
    return PipelineResult(success=True, data={"generated_text": "Not implemented yet"})